### PR TITLE
Imagerev & Fontrev

### PR DESCRIPTION
**Pull request by jbmoelker**
Create a copy of each image or font file with a unique filename. The filename is based on the original filename with `/-fastatic-[0-9a-f]{5}/` added to the basename. This allows these files to be cached forever:

screen shot 2016-08-26 at 12 47 21

Note: The original files are not removed as there may still be references to it (e.g. in a script w/ string concatenation).

The stats show a negative improvement as while the image and font files get smaller, they also double up:

screen shot 2016-08-26 at 12 47 30
screen shot 2016-08-26 at 13 29 24

I think this just means our stats generator has to work differently: compare files in src with the same files in dest instead of all files in dest. @petergoes should we make a separate branch/PR to fix that in the stats generator before allowing this PR to be merged?

### CODE REVIEW

**Comment by petergoes** [comment]
Yeah, I think the representation of the stats is not related to this PR. I am refactoring the stats at the moment. So that is in progress

**Comment by jpsc** [comment]
relative to #9 ?

File: lib/parse-rev.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			rename(file => file.basename = prefixHash(file.basename)),
+			gulp.dest(config.cwd),
+			rev.manifest((config.name || 'rev') + '-manifest.json'),
+			gulp.dest(config.cwd)
+		], (err) => err ? reject(err) : resolve())
+	});
+}
+
+/**
+ * Replaces hash (10 character alphanumeric [a-f0-9]) with `-fastatic-{hash}` (first 5 chars of original hash).
```
Comment (petergoes) [inline lib/parse-rev.js:35]: Why do we only use the first 5 chars of the hash? What is the harm in 10 chars?

**Comment by petergoes** [comment]
@jbmoelker What is the reason we only rev images and font files? can't we rev js and css files as well?

**Comment by jbmoelker** [comment]
@petergoes yes we can. But I think we need to do that in separate PRs. Image and font files can be referenced in HTML, CSS and JS files. So those references first need to be replaced (which is also an issue in itself) before HTML, CSS and JS can be revisioned.

**Pull request merged**